src/deadline_worker_agent/scheduler/scheduler.py

```diff
@@ -668,6 +669,98 @@ def _fail_all_actions(
+        # Raises a ValueError if an impossible situation arises and we need to fail the Session.
+        os_user: Optional[SessionUser] = None
+        if not job_run_as_user_override.run_as_agent:
+            if job_run_as_user_override.job_user is not None:
+                os_user = job_run_as_user_override.job_user
+                logger.info(
+                    SessionLogEvent(
+                        subtype=SessionLogEventSubtype.USER,
+                        queue_id=queue_id,
+                        job_id=job_id,
+                        session_id=session_id,
+                        user=os_user.user,
+                        message="Running as host-configured override user.",
+                    )
+                )
+            elif job_run_as_user is None:
+                # Terminal error. We need to fail the Session.
+                # This should *never* happen; it occuring would mean that a service invariant has
+                # been violated.
+                message = (
+                    "FATAL: Queue does not have a jobRunAsUser. This should not be possible. "
+                    "Please report this to the service team."
+                )
+                raise ValueError(message)
+            elif not job_run_as_user.is_worker_agent_user:
+                # If we do not have a job-user override & we're not explicitly running
+                # as the agent's user, then we *MUST* have a jobRunAsUser from the JobDetails.
+                # Reasons:
+                #  1) The service always allows service-managed Fleets to associate with
+                #     a Queue. The SMF Worker Agent is *always* run with a local user override.
+                #  2) The service only allows a customer-managed Fleet to associate with a
+                #     Queue if either:
+                #       a) The jobRunAsUser is explicitly set to WORKER_AGENT_USER; or
+                #       b) The jobRunAsUser is explicitly set to QUEUE_CONFIGURED_USER and
+                #          a user has been defined for the CMF's OS Platform (Linux/MacOS Fleets must
+                #          have a "posix" user; and Windows Fleets must have a "windows" user)
+                #  3) The service does not allow a Queue's jobRunAsUser to be updated if the constraint
+                #     imposed by (2) above would be violated for one or more of that Queue's current QFAs.
+                if host_is_posix:
+                    os_user = job_run_as_user.posix
+                else:
+                    os_user = job_run_as_user.windows
+                if os_user is None:
+                    # Terminal error. We need to fail the Session.
+                    # This should *never* happen; it occuring would mean that a service invariant has
+                    # been violated.
+                    message = (
+                        "FATAL: Queue's jobRunAsUser does not define a QUEUE_CONFIGURED_USER for this platform. "
```

Comment:
Should we include which platform in the message for helping diagnose? I suppose the agent logs will include the platform in the startup logging, right?

Reply:
The platform is indeed already in the log.